Improve infra layer

gustavorodarte opened the pull request.

gustavorodarte committed:
chore: add Mongose Hero Mapper

gustavorodarte committed:
test(factory): add factory-Bot loader

gustavorodarte committed:
test: implemented hero factory and safeDatafaker with Chance

gustavorodarte committed:
chore(package,json): resolve conflicts Package.json

gustavorodarte committed:
feat(logger.js): implemented Winston at default logger

gustavorodarte committed:
feat(infra/loggin/looger.js): added Winston to logger

clsechi commented on specs/support/cleanDatabase.js:
```diff
@@ -0,0 +1,29 @@
+  
+  if (database.connection.readyState === 0) {
+    await database.connect(
+      `${process.env.MONGO_URL}${process.env.TEST_SUITE}`,
```
use env-var

clsechi commented on specs/setup.js:
```diff
@@ -0,0 +1,18 @@
+const container = require('src/container');
+const database = container.resolve('database');
+
+const cleanDatabase = require('specs/support/cleanDatabase');
```
usar no beforeEach dos testes de precisam disso

clsechi commented on specs/unit/domain/hero/Hero.module.spec.js:
```diff
@@ -0,0 +1,37 @@
+const { Ok, Err } = require("crocks/Result");
+const { HeroDomainFactory } = require('../../../../src/domain/hero');
```
usar node_path

clsechi commented on src/infra/repositories/hero/MongooseHeroMapper.js:
```diff
@@ -0,0 +1,33 @@
+
+const MongooseHeroMapper = {
+  toDomainObject({
+    superPowers,
```
o ideal era usar o `toObject` to mongoose para remover qualquer metodo dele

clsechi commented on .example.env:
```diff
@@ -9,3 +9,4 @@ MONGO_DB_URL=mongodb://mongo:27017/application
 NODE_ENV=development
 NODE_PATH=.
 PORT=3000
+NODE_PATH=.
```
duplicado

gustavorodarte committed:
feat(interface layer): wip implement rest interface

gustavorodarte committed:
feat(interface layer): wIP: Implement rest interface

:wq

clsechi committed:
[ci-skip] WIP

clsechi committed:
[ci-skip] WIP

gustavorodarte committed:
test(crocksmatcher.js): implement jest custom matcher for Crocks Result

gustavorodarte committed:
feat(herocontroller): implemented Hero controller in rest interface

gustavorodarte committed:
fix(herocontroller): fixed setup of express rest controllers

gustavorodarte committed:
refactor(graphql resolvers): refactor graphql resolver to working with new operation output

clsechi committed:
WIP

clsechi committed:
Merge remote-tracking branch 'origin/master' into improve_infra_layer

gustavorodarte committed:
feat(logger.middleware): add rule fort morgan ignore graphql playground requests

gustavorodarte committed:
fix(errorhandler): fix and create error handlers for Graphql and Rest Interfaces

gustavorodarte commented on src/infra/factoryBot/FactoriesLoader.js:
```diff
@@ -0,0 +1,36 @@
+}) => {
+  // eslint-disable-next-line fp/no-unused-expression
+  loadFactories(baseFolder)
+    .map((file) => {
```
I think we can refactor this piece of code to a reduce, and return the factoryBot in the end.

gustavorodarte reviewed (commented):
needed some refactor to match with the functional approach,

gustavorodarte committed:
test: remove unfinished tests

gustavorodarte committed:
fix(root.router): disable contentSecurityPolicy header on development

Helmet contenSecurityPolicy don't allow the apollo playground works, so this commit removes in dev
env

clsechi commented on specs/setup.js:
```diff
@@ -0,0 +1,24 @@
+  toEqualOk,
+});
+
+beforeEach(async (done) => cleanDatabase(database, done));
```
![really](https://media.makeameme.org/created/oh-really-do-cbffsw.jpg)

clsechi commented on specs/support/factories/hero.js:
```diff
@@ -0,0 +1,49 @@
+  minimize: false,
+  versionKey: false,
+  /* eslint-disable no-underscore-dangle, no-param-reassign */
+  transform: (doc, ret) => {
```
nos falamos d manter o funcional somente na camada de aplicação

gustavorodarte replied:
Não, o que foi falado é que seriamos menos rigorosos com a extremidades da aplicação. Mas isso não siguinifica que vamos abandonar totalmente o paradigma funcional. Ainda mais em casos com esse, que é facilmente resolvido.

clsechi replied:
vejo ganho zero em fazer isso para aplicação final sem si, ja não vamos usar funcional nos testes, isso pode ser feito futuramente no boilerplate em sí,  deixando ele 100% funcional

clsechi commented on specs/unit/infra/hero/MongooseHeroRepository.spec.js:
```diff
@@ -0,0 +1,74 @@
+
+      test('should return the created hero', async () => {
+        const result = await HeroRepository.add(payload).toPromise();
+        expect(resultToObject(result).ok).toStrictEqual({
```
pq os objetos não são iguais, eu tenho um id no objeto q vem do banco

gustavorodarte committed:
fix: fix review request changes

clsechi committed:
feat: add repository methods with specs, factories and other minnor changes

clsechi committed:
ci: add verbose mode to env-cmd

gryzinsky committed:
ci: add MONGO_DB_URL to GitHub CI test job

gustavorodarte committed:
Merge branch 'improve_infra_layer' of github.com:zrpaplicacoes/node-functional-boilerplate into improve_infra_layer

gustavorodarte committed:
Merge pull request #26 from zrpaplicacoes/rest_interface

Add Rest Interface
